src/ripple/nodestore/impl/Shard.h

```diff
@@ -316,29 +318,30 @@ class Shard final
     boost::filesystem::path const dir_;
 
     // Storage space utilized by the shard
-    std::uint64_t fileSz_{0};
+    GUARDED_BY(mutex_) std::uint64_t fileSz_{0};
 
     // Number of file descriptors required by the shard
-    std::uint32_t fdRequired_{0};
+    GUARDED_BY(mutex_) std::uint32_t fdRequired_{0};
```

Comment:
If you move the initialization to the constructor, can you place the annotation at the end of this declaration?

Reply:
Yes - but while I'm not happy about putting the annotation at the beginning, I'd rather do that than disallowing initializing in the declaration